### ISSUE TITLE
Admin: inconsistent sorting behavior

### ISSUE THREAD

**Issue by vohmar** [internetee/auction_center#400]
Related to: #321 

Similarly to auctions view other admin views suffer from some unexpected sorting behavior as well

* /admin/results
  * "doman name", "finished at" and "registration due date" override sorting selected on other columns - status, name. Example:
 set ascending order on column 'status':
![image](https://user-images.githubusercontent.com/7734747/68199114-0db3b780-ffc6-11e9-86ef-39671823c7c4.png)

  * now trying to sort the resulting list by 'registration due date' in descending order
![image](https://user-images.githubusercontent.com/7734747/68199277-671be680-ffc6-11e9-8861-cbe7e452eeb3.png)

  * while the other way around. Reset the list and sorted domains by 'finished at' desc
![image](https://user-images.githubusercontent.com/7734747/68199534-e1e50180-ffc6-11e9-9a18-3d70b8143720.png)

  * and now sorting the results by 'name' asc the pervious sorting order is preserved
![image](https://user-images.githubusercontent.com/7734747/68200751-296c8d00-ffc9-11e9-9693-b0b062ac3df7.png)

* /admin/auctions
  * sililarly to results view - 'domain name' and 'ends at' columns override periously set order while the other columns do not do that.

* /admin/billing_profiles
  * column 'name' overrides all previous sorting 

* /admin/users
  * column 'email' overrides previously set sorting
  * sorting option on column 'roles' is missing

* /admin/bans
  * nothing overrides  'domain name' and 'invoice' and these columns override everything except each other
  * everything overrides 'user'
  * 'valid until' is overriden by 'valid from' in addition to 'domain name' and 'invoice'

* /admin/invoices
  * column 'billing profile' overrides all other column orders
  * columns 'due date' and 'status' override all but 'billing profile' sorting orders
  * sorting by currency columns does not work - 'total', 'total paid'

**Comment by yulgolem** [comment]
The main issue here is following:
On sorting we are totally dependent on order of parameters in URL. We are assuming what then we are sorting by one column and this column is added un URL params, the second sorting column will be added after the first... and we are wrong.

The URL generation uses Hash#to_params method, which sorts parameters alphabetically. This is a known Rails feature [since 2011](https://github.com/rails/rails/issues/1146). So - then we are adding second, third etc sorting parameter, they are sorted in URL (and in dataset query) not by time of adding, but by their alphabetic order.

The solution is to sort datasets only by one column at a time, removing previous sorting params from URL.

**Comment by vohmar** [comment]
agreed, lets solve the confusing situation by sorting one column at a time.

**Comment by vohmar** [comment]
Invoices - total and total paid columns don't sort

**Comment by yulgolem** [comment]
> Invoices - total and total paid columns don't sort

There is an issue here - atm we cannot sort on invoice total because it's not a model field, but calculated. I can remove sorting buttons from this column header.

But sorting by total paid is a go, I've added.

**Comment by vohmar** [comment]
spotted some more sorting issues - probably were there in the last go as well just did not register these.

* text sorting does not seem to support estonian alphabet ie letter õ is places with o. In estonian alphaet letters õ, ä, ö, ü are at the end (https://en.wikipedia.org/wiki/Estonian_orthography)
* Results - name: the records seem to be ordered by id not the name
* Billing profile - user: not sure what is going on here
![image](https://user-images.githubusercontent.com/7734747/69650077-b78edd00-1076-11ea-89eb-935ea68f58ef.png)
* Users - username: same as the previous point as this uses the same db field for sorting
* Invoices - status: it appears that the statuses are sorted in the opposite order
![image](https://user-images.githubusercontent.com/7734747/69650522-64695a00-1077-11ea-92d7-2d8c83ed91fe.png)

**Comment by yulgolem** [comment]
Hmmm... and here I've got an idea.
You see, all our sorting inconveniences comes from the fact what we are using server-side sorting - and we are sorting not the table content, but dataset for this table. And sorting is based on data stored, not on data we see.

For example, for `Billing profile - user`, as far as I see, we are sorting by database column user.surname, because this dataset is getting this column to fill in the table.

Same with `Invoices - status`, the statuses of the invoice in DB sorted not by alphabetical order, but in order of definition in model enumeration, and `cancelled` is a last status defined.

And same with the alpha-order of strings, DB is set on En locale, not on Et, so order by fields with estonian letters processed as if were just english.

Maybe we should think about swapping server-side ordering to browser-side and find some table-components allowing JS-sorting? We are already using Semantic UI in our front, so maybe https://semantic-ui.com/collections/table.html#sortable will do the trick?

**Comment by vohmar** [comment]
figured the billing profile and users table sorting issue - failed to notice that names are sorted by surename, but are displayed first name first so technically no problem there might call for some UI change though to that the behaviour would be obvious to users as well ie display names surename first as well.

Will browser side sorting work with paginated views - browsers do not have full dataset?

merged the update that remoces mutli column sorting: https://github.com/internetee/auction_center/pull/408

other issues remain to be solved:
* text sorting to support Estonian alphabet
* results/name column ordered by id instead of name
* invoice/status ordered by id instead of status name

**Comment by vohmar** [comment]
sorting in staging with the proposed solution renders strange results
![image](https://user-images.githubusercontent.com/7734747/73460172-b634a600-4380-11ea-8f50-2d04987d6630.png)

the first record timo-ajalugu-1.ee never moves from the first position, rest are sorting correctly.

results seem to order correctly, although there is the name column sorting issue by the last name. SImilar name sorting issue is also on billing profile (user, naem), users

invoice statuses do no order correctly - cancelled, paid, issued

this all on sigle page views with small amount of data

**Comment by vohmar** [comment]
works well, the lag of getting the list of 500 records is tolerable and there are no users that have more than 100 records yet.

Lets try to remove pagination as the full list is in browser anyway - does rendering the full list still keep the user experience tolerable for admin and more active users.

**Comment by OlegPhenomenon** [comment]
@vohmar sort issue still actual on current blind auction?
